keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/DeliveryConfig.kt

```diff
@@ -16,11 +19,16 @@ data class SubmittedDeliveryConfig(
   val environments: Set<SubmittedEnvironment> = emptySet()
 )
 
+@JsonDeserialize(using = SubmittedEnvironmentDeserializer::class)
 data class SubmittedEnvironment(
   val name: String,
   val resources: Set<SubmittedResource<*>>,
   val constraints: Set<Constraint> = emptySet(),
-  val notifications: Set<NotificationConfig> = emptySet()
+  val notifications: Set<NotificationConfig> = emptySet(),
+  /**
+   * Optional locations that are propagated to any [resources] where they are not specified.
+   */
+  val locations: SubnetAwareLocations? = null
```

Comment:
New property added on `SubmittedEnvironment`. Note there isn't a corresponding one on `Environment`, we just throw it away after applying to the resources.